test/functional/rpc_getblockstats.py

```diff
@@ -70,90 +86,63 @@ def run_test(self):
         self.sync_all()
         node.generate(1)
 
-        start_height = 101
-        max_stat_pos = 2
-        stats = [node.getblockstats(height=start_height + i) for i in range(max_stat_pos+1)]
+        self.expected_stats = self.get_stats()
+
+        blocks = []
+        tip = node.getbestblockhash()
+        blkhash = None
+        height = 0
+        while tip != blkhash:
+            blkhash = node.getblockhash(height)
+            blocks.append(node.getblock(blkhash, 0))
+            height += 1
+
+        with open(filename, "w") as f:
+            f.write("""{\n  "stats": """)
+            json.dump(self.expected_stats, f, indent=4)
+            f.write(""",\n  "blocks": """)
+            json.dump(blocks, f)
+            f.write("""\n}\n""")
+
+    def load_test_data(self, filename):
+        node = self.nodes[0]
+
+        with open(filename, "r") as f:
+            d = json.load(f)
+            blocks = d["blocks"]
+            self.expected_stats = d["stats"]
+            del d
+
+        for b in blocks:
+            node.submitblock(b)
+
+    def get_stats(self):
+        node = self.nodes[0]
+        return [node.getblockstats(height=self.start_height + i) for i in range(self.max_stat_pos+1)]
+
+    def run_test(self):
+        test_data = os.path.join(TESTSDIR, self.options.test_data)
+        if self.options.gen_test_data:
+            self.generate_test_data(test_data)
+        else:
+            self.load_test_data(test_data)
+
+        node = self.nodes[0]
+        stats = self.get_stats()
 
         # Make sure all valid statistics are included but nothing else is
         assert_equal(set(stats[0].keys()), set(self.EXPECTED_STATS))
 
-        print(stats)
```

Comment:
commit: Tests: Save and load block and corresponding expected statistics

If all these test cases are getting removed anyway, I'd just drop them from the previous commit.